canvasapis.py

```diff
@@ -0,0 +1,150 @@
+                  '/enrollments?per_page=100&type[]=StudentEnrollment'
+        try:
+            response = self.make_api_call(url, self.HTTP_METHOD_GET)
+        except:
+            raise
+
+        return response
+
+    def create_group(self, group_category_id, group_name, course_id):
+        """
+        Creates a group in course under a particular group category, In case of exceptions it throws it to calling method
+        :param String group_category_id:
+        :param String group_name:
+        :return:
+        """
+
+        url = self.canvas_url + '/api/v1/groups?course_id=' + course_id + '&group_category_id=' \
+              + group_category_id + '&name=' + urllib.quote_plus(group_name)
+
+        try:
+            response = self.make_api_call(url, self.HTTP_METHOD_POST)
+        except:
+            raise
+
+        return response
+
+    def get_next_page_url(self, response):
+        """
+        get the next page url from the Http response headers
+        :param HttpResponse response:
+        :return: String next_page_url
+        """
+        logging.debug(self.get_next_page_url.__name__ + '() called')
+
+        if not response.links:
+            logging.debug('The api call do not have Link headers')
+            return None
+
+        for page in response.links:
+            if 'next' in page:
+                return response.links['next']['url']
+
+    def make_api_call(self, url, http_method):
+        """
+        bubbles up the exception to be caught and handled by original calling function
+        :param String url:
+        :param String http_method:
+        :return: HttpResponse response
+        """
+        try:
+            response = self.api_handler(url, http_method)
+        except:
+            raise
+
+        return response
+
+    def api_handler(self, url, request_type):
+
+        """
+        This is the generic api handler for making HTTP calls, will throw a RequestException which encompass
+        all of the exceptions that might go wrong in making api call
+        More info, https://raw.githubusercontent.com/kennethreitz/requests/master/requests/exceptions.py
+        http://docs.python-requests.org/en/master/user/quickstart/#errors-and-exceptions
+        :param String url:
+        :param HttpMethod request_type:
+        :return: HttpResponse api_response
+        """
+        logging.debug(self.api_handler.__name__ + '() called')
+        logging.info('URL: ' + url)
+        response = None
+        headers = {self.CONTENT_TYPE: self.MIME_TYPE_JSON, self.AUTHORIZATION: self.BEARER + self.canvas_token}
+        try:
+            if request_type == self.HTTP_METHOD_GET:
+                response = requests.get(url, headers=headers)
+            elif request_type == self.HTTP_METHOD_POST:
+                response = requests.post(url, headers=headers)
+        except requests.exceptions.RequestException as request_exception:
+            raise request_exception
+
```

Comment:
Consider logging, for every request, if there is a link header returned.  Assuming you know which requests may or may not return multiple pages is ok, but it would be good to have some indication if there is data that might have been ignored.